### PR TITLE
Lint and format Tools/build/check-warnings.py

### PR DESCRIPTION
**Pull request by hugovk**
cc @nohlson @sethmlarson

### CODE REVIEW

File: Tools/build/check_warnings.py [hunk 1]
```diff
@@ -126,7 +128,8 @@ def is_file_ignored(
     file_path: str, ignore_rules: set[IgnoreRule]
 ) -> IgnoreRule | None:
     """
-    Returns the IgnoreRule object for the file path if there is a related rule for it
+    Returns the IgnoreRule object for the file path
+    if there is a related rule for it
     """
```
Comment (AlexWaygood) [inline Tools/build/check_warnings.py]: nit: it's nice for the first sentence of a docstring to be short. Maybe this?

```py
    """Return the IgnoreRule object for the file path.

    Return ``None`` if there is no related rule for that path.
    """
```

**Review by AlexWaygood — APPROVED** [review]

File: Tools/build/check_warnings.py [hunk 2]
```diff
@@ -4,9 +4,9 @@
 """
 
 import argparse
-from collections import defaultdict
 import re
 import sys
+from collections import defaultdict
```
Comment (AA-Turner) [inline Tools/build/check_warnings.py:9]: If you make the following change, this `defaultdict` import can be removed:

```diff
-warnings_by_file = defaultdict(list)
+warnings_by_file = {}
```

```diff
-warnings_by_file[warning["file"]].append(warning)
+warnings_by_file.setdefault(warning["file"], []).append(warning)
```

```suggestion
```

File: Tools/build/check_warnings.py [hunk 3]
```diff
@@ -191,7 +194,10 @@ def get_unexpected_improvements(
     """
     unexpected_improvements = []
     for rule in ignore_rules:
-        if not rule.ignore_all and rule.file_path not in files_with_warnings.keys():
+        if (
+            not rule.ignore_all
+            and rule.file_path not in files_with_warnings.keys()
```
Comment (AA-Turner) [inline Tools/build/check_warnings.py:199]: This function is typed incorrectly, `files_with_warnings: set[IgnoreRule],` needs to be `dict[...]` or it should take `files_with_warnings.keys()` for a set-like interface (in which case you would need to use `Set[...]` from `collections.abc`.
```suggestion
            and rule.file_path not in files_with_warnings
```

**Review by AA-Turner — APPROVED** [review]
@nohlson I've left a few pointers which might be of interest.

A

File: Tools/build/check_warnings.py (hunk 2 above)
Comment (AlexWaygood) [reply Tools/build/check_warnings.py:9]: I generally prefer `defaultdict`; it's more concise at the point of use and faster. But I don't have a strong opinion :-)

**Pull request merged**